test/models/domain_test.rb

```diff
@@ -8,4 +8,25 @@ class DomainTest < ActiveSupport::TestCase
     domain = Domain.new(name: "foo.com", whois_response: "WHOIS DATA")
     assert_not_nil domain.populate 
   end
+
+  # incorrect domain name raises error
+  test "populate with invalid domain name sholud raise error" do
+  	domain = Domain.new(name: "foocom", whois_response: "WHOIS DATA")
+  	assert_raise do
+  		domain.populate
+  	end
+  end
+
+  # This test is intended to fail to indicate that future work is needed to treat these domains equally
+  test "These domains should be treated equally" do
+  	domain = Domain.new(name: "foo.com", whois_response: "WHOIS DATA")
+  	domain2 = Domain.new(name: "www.foo.com", whois_response: "WHOIS DATA")
+  	assert_equal(domain.name,domain2.name)
+  end
+
+  # check regex if valid domain name is given 
```

Comment:
I think validating that the domain name we pass in has a dot in it is a good idea in general, but this only does so in the tests. Is there a way that we can test this validitiy when we create the domain?